tests/MongoDB.Driver.Tests/Specifications/unified-test-format/UnifiedValueMatcher.cs

```diff
@@ -0,0 +1,234 @@
+﻿/* Copyright 2020-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using FluentAssertions;
+using MongoDB.Bson;
+using MongoDB.Bson.IO;
+using MongoDB.Bson.TestHelpers.XunitExtensions;
+using Xunit.Sdk;
+
+namespace MongoDB.Driver.Tests.Specifications.unified_test_format
+{
+    public class UnifiedValueMatcher
+    {
+        private EntityMap _entityMap;
+
+        public UnifiedValueMatcher(EntityMap entityMap)
+        {
+            _entityMap = entityMap;
+        }
+
+        public void AssertValuesMatch(BsonValue actual, BsonValue expected)
+        {
+            var jsonWriterSettings = new JsonWriterSettings { Indent = true };
+            try
+            {
+                AssertValuesMatch(actual, expected, isRoot: true);
+            }
+            catch (XunitException exception)
+            {
+                throw new AssertionException(
+                        $"{Environment.NewLine}Expected value to be: {expected?.ToJson(jsonWriterSettings)}" +
+                        $"{Environment.NewLine}But found: {actual?.ToJson(jsonWriterSettings)}",
+                    exception);
+            }
+        }
+
+        private void AssertValuesMatch(BsonValue actual, BsonValue expected, bool isRoot)
+        {
+            if (expected.IsBsonDocument &&
+                expected.AsBsonDocument.ElementCount == 1 &&
+                expected.AsBsonDocument.GetElement(0).Name.StartsWith("$$"))
+            {
+                var specialOperatorDocument = expected.AsBsonDocument;
+                var operatorName = specialOperatorDocument.GetElement(0).Name;
+                var operatorValue = specialOperatorDocument[0];
+
+                switch (operatorName)
+                {
+                    case "$$type":
+                        AssertExpectedType(actual, operatorValue);
+                        break;
+                    case "$$matchesHexBytes":
+                        AssertValuesMatch(actual, operatorValue, true);
+                        break;
+                    case "$$unsetOrMatches":
+                        if (actual == null)
+                        {
+                            break;
+                        }
+                        AssertValuesMatch(actual, operatorValue, true);
+                        break;
```

Comment:
Simplify 70-76 to:

    case "$$unsetOrMatches":
        if (actual != null)
        {
            AssertValuesMatch(actual, operatorValue, true);
        }
        break;

Reply:
i.e. only one `break` at end of `case`